addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -1,27 +1,415 @@
+                # Gather information
+                parsed_update_data = self._l10n_it_edi_parse_update_data(updates_data[id_transaction], move_to_update)
+                transformed_update_data = self._l10n_it_edi_transform_update_data(parsed_update_data, id_transaction, move_to_update)
+                message = self._l10n_it_edi_get_message(move_to_update, transformed_update_data)
+
+                # Process the update
+                if id_transaction_to_ack := self._l10n_it_edi_process_update(
+                    move_to_update, id_transaction, transformed_update_data, message):
+                    acks.append(id_transaction_to_ack)
+
+            # Send the acks
+            if acks:
+                self._make_request(f'{server_url}/api/l10n_it_edi/1/ack', params={'transaction_ids': acks})
+
+        except AccountEdiProxyError as pe:
+            raise UserError(_("An error occurred while contacting the Proxy Server: (%s) %s", pe.code, pe.message))
+        except Exception as e:
+            raise UserError(_("An error occurred while updating the invoice(s) from the Proxy Server: %s", e))
+
+    def _l10n_it_edi_parse_update_data(self, update_data, move_to_update):
+        sdi_state = update_data.get('state', '')
+        encrypted_update_content = update_data.get('file')
+        encryption_key = update_data.get('key')
+        if not (encrypted_update_content and encryption_key):
+            return {'sdi_state': sdi_state}
+
+        decrypted_update_content = etree.fromstring(self._decrypt_data(encrypted_update_content, encryption_key))
+        outcome = get_text(decrypted_update_content, './/Esito')
+        filename = get_text(decrypted_update_content, './/NomeFile')
+        date_arrival = get_datetime(decrypted_update_content, './/DataOraRicezione') or fields.Date.today()
+
+        errors = [(
+            get_text(error_element, '//Codice'),
+            get_text(error_element, '//Descrizione'),
+        ) for error_element in decrypted_update_content.xpath('//Errore')]
+
+        return {
+            'sdi_state': sdi_state,
+            'errors': errors,
+            'outcome': outcome,
+            'date': date_arrival,
+            'filename': filename,
+        }
+
+    def _l10n_it_edi_transform_update_data(self, parsed_update_data, id_transaction, move_to_update):
+        """ Reads the notification XML coming from the EDI Proxy Server
+            Recovers information about the new state.
+            Computes whether the EDI Proxy Server is to be acked,
+            and whether the id_transaction has to be reset.
+        """
+        update_state_map = {
+            'not_found': False,
+            'awaiting_outcome': 'processing',
+            'notificaScarto': 'rejected',
+            'ricevutaConsegna': 'forwarded',
+            'forward_attempt_failed': 'forward_attempt_failed',
+            'notificaMancataConsegna': 'forward_failed',
+            ('notificaEsito', 'EC01'): 'accepted_by_pa_partner',
```

Comment:
It doesn't. Simply, the key used to get a new state is not always a string, it can be a tuple.
https://github.com/odoo/odoo/pull/122194/files/d357301594fd31ec52472b0c98bf9d606adbc1be#diff-9f510dc6027ee67d2d6985ade31d1ce8f1bab382165ff588a68bef9031636beeR299